types/scheduler.d.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Task priorities that determine the order in which tasks run.
+ *
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Prioritized_Task_Scheduling_API#task_priorities)
+ */
+export type TaskPriority = 'user-blocking' | 'user-visible' | 'background';
+
+/**
+ * {@link Scheduler.postTask} options.
+ *
+ * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Scheduler/postTask#options)
+ */
+export type SchedulerPostTaskOptions = {
+  /** The immutable {@link TaskPriority} of the task. One of `"user-blocking"`, `"user-visible"`, or `"background"`. If set, this priority is used for the lifetime of the task and priority set on the `signal` is ignored. */
+  priority?: TaskPriority;
+  /** An {@link AbortSignal} or {@link TaskSignal} that can be used to abort or re-prioritize the task (from its associated controller). */
+  signal?: AbortSignal | TaskSignal;
+  /** The minimum amount of time after which the task will be added to the scheduler queue, in whole milliseconds. The actual delay may be higher than specified, but will not be less. The default delay is 0. */
+  delay?: number;
+};
```

Comment:
@shaseley this is where I had some question about how IDL works. e.g. `postTask`'s `options` parameter is marked `optional`, but the individual properties on `SchedulerPostTaskOptions` don't have a similar annotation. Are `dictionary` members all optional by default? https://wicg.github.io/scheduling-apis/#schedule-a-posttask-task has clear fallbacks for `signal` and `priority` aren't defined, but there's no such language for `delay`

Reply:
ok, should have read for more than 30 seconds.

> Are dictionary members all optional by default?

Looks like that's exactly the case: https://webidl.spec.whatwg.org/#es-dictionary, and delay has a default value of 0